web/src/components/shared/saleItem/SaleItem.tsx

```diff
@@ -16,7 +16,7 @@ function showStars(star: number) {
     return <p>{stars}</p>;
 }
 
-export const SaleItem = ({ img, name, price, stars }: Product) => {
+export const SaleItem = ({ img, name, price, stars }: ProductSale) => {
```

Comment:
this can be Product. 
You dont need to create class. It allows us to extend this component by other Product properties without changing its type.